### PR TITLE
Added example of regex checking for IAM arns

### PR DESCRIPTION
**Pull request by wey-chiang**
### Background

Added example of regex checking for IAM arns

### Testing

Manual testing in a python shell

### CODE REVIEW

File: aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.py [hunk 1]
```diff
@@ -26,5 +33,10 @@ def rule(event):
         return True
 
     # Only approved IAM Roles can make IAM Changes
-    return event['userIdentity']['sessionContext']['sessionIssuer'][
-        'arn'] not in IAM_ADMIN_ROLES
+    for admin_role_pattern in IAM_ADMIN_ROLES:
```
Comment (nhakmiller) [inline aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.py:36]: Should this be `for admin_role_pattern in IAM_ADMIN_ROLE_PATTERNS:` ?

Comment (wey-chiang) [reply aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.py:34]: Doh, thanks for that catch

File: aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.py [hunk 2]
```diff
@@ -1,30 +1,48 @@
...
+        if (len(
+                re.findall(
+                    admin_role_pattern,
+                    event["userIdentity"]["sessionContext"]["sessionIssuer"]
+                    ["arn"],
+                )) > 0):
+            return False
+
+        return (event["userIdentity"]["sessionContext"]["sessionIssuer"]["arn"]
+                not in IAM_ADMIN_ROLES)
```
Comment (nhakmiller) [inline aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.py:45]: probably need to  drop  these two  lines as well (also I know it's late there feel free to finish this up in the morning).

File: aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.py [hunk 3]
```diff
@@ -1,30 +1,48 @@
...
-        'arn'] not in IAM_ADMIN_ROLES
+    for admin_role_pattern in IAM_ADMIN_ROLE_PATTERNS:
+        # Check if the arn matches any role patterns, returns False (whitelisting it) if there is a match
+        if (len(
+                re.findall(
+                    admin_role_pattern,
+                    event["userIdentity"]["sessionContext"]["sessionIssuer"]
+                    ["arn"],
+                )) > 0):
+            return False
```
Comment (lindsey-w) [inline aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.py:42]: Are there some logs we could pull from for testing the new logic?

Comment (wey-chiang) [reply aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.py:42]: Created a new test for this

File: aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.yml [hunk 4]
```diff
@@ -106,6 +106,48 @@ Tests:
           "eventType": "AwsApiCall",
           "recipientAccountId": "123456789012"
       }
+  -
+    Name: IAM Entity Created Manually With Approved Role Pattern
+    ExpectedResult: true
```
Comment (lindsey-w) [inline aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.yml:111]: Should this be false?

**Review by lindsey-w — APPROVED** [review]
Nice! Just had one question about the test case.

**Review by nhakmiller — APPROVED** [review]

File: aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.yml (hunk 4 above)
Comment (wey-chiang) [reply aws_cloudtrail_rules/aws_iam_entity_created_without_cloudformation.yml]: Yup, debugged it to find that I needed the `invokedBy: "cloudformation.amazonaws.com"` in the example, added that and the test started working as intended

**Review by nhakmiller — APPROVED** [review]

**Pull request merged**